lib/temporal/client/workflow_handle.rb

```diff
@@ -3,8 +3,51 @@
 
 module Temporal
   class Client
+    # Handle for interacting with a workflow.
+    #
+    # This is usually created via {Temporal::Client#workflow_handle} or returned
+    # from {Temporal::Client#start_workflow}.
     class WorkflowHandle
-      attr_reader :id, :run_id, :result_run_id, :first_execution_run_id
+      # @return [String] ID for the workflow.
+      attr_reader :id
+
+      # Run ID used for {#signal} and {#query} calls if present to ensure the query or signal happen
+      # on this exact run.
+      #
+      # This is only created via {Temporal::Client#workflow_handle}.
+      # {Temporal::Client#start_workflow} will not set this value.
+      #
+      # This cannot be mutated. If a different run ID is needed, {Temporal::Client#workflow_handle}
+      # must be used instead.
+      #
+      # @return [String]
+      attr_reader :run_id
+
+      # Run ID used for {#result} calls if present to ensure result is for a workflow starting from
+      # this run.
+      #
+      # When this handle is created via {Temporal::Client#workflow_handle}, this is the same as
+      # `:run_id`. When this handle is created via {Temporal::Client#start_workflow}, this value
+      # will be the resulting run ID.
+      #
+      # This cannot be mutated. If a different run ID is needed, {Temporal::Client#workflow_handle}
+      # must be used instead.
+      #
+      # @return [String]
+      attr_reader :result_run_id
+
+      # Run ID used for {#cancel} and {#terminate} calls if present to ensure the cancel and
+      # terminate happen for a workflow ID started with this run ID.
+      #
+      # This can be set when using {Temporal::Client#workflow_handle}. When
+      # {Temporal::Client#start_workflow} is called without a start signal, this is set to the
+      # resulting run.
+      #
+      # This cannot be mutated. If a different first execution run ID is needed,
+      # {Temporal::Client#workflow_handle} must be used instead.
+      #
+      # @return [String]
+      attr_reader :first_execution_run_id
 
       def initialize(client_impl, id, run_id: nil, result_run_id: nil, first_execution_run_id: nil)
```

Comment:
This is private, right?

Reply:
I can't make it private since it's used by another class (`Client`)